# Renames Ruby's `lgamma_r` polyfill to `mkxp_ruby_lgamma_r`.
# Using `lgamma_r` as the name causes the linker error "undefined reference to `lgamma_r'" when building for Windows using GCC if link-time optimization is enabled.

--- a/include/ruby/missing.h
+++ b/include/ruby/missing.h
@@ -114,8 +114,8 @@ RUBY_EXTERN double erfc(double);
 #ifndef HAVE_TGAMMA
 RUBY_EXTERN double tgamma(double);
 #endif
-
 #ifndef HAVE_LGAMMA_R
+#define lgamma_r mkxp_ruby_lgamma_r
 RUBY_EXTERN double lgamma_r(double, int *);
 #endif
 
